packages/flutter/lib/src/material/switch.dart

```diff
@@ -350,7 +356,7 @@ class _SwitchState extends State<Switch> with TickerProviderStateMixin {
             activeTrackColor: activeTrackColor,
             inactiveTrackColor: inactiveTrackColor,
             configuration: createLocalImageConfiguration(context),
-            onChanged: widget.onChanged,
+            onChanged: enabled ? _handleValueChanged : null,
```

Comment:
_RenderSwitch.onChanged and _SwitchRenderObjectWidget.onChanged should probably also be changed to `ValueChanged<bool>?` to indicate that the value here can never be null. After that, we should just pass widget.onChanged down unchanged and add the logic from _handleValueChanged to one of the lower levels.

Reply:
Yeah, I wasn't sure how far to take this change, so I stuck with just the public facing API, as that was the important one. I will look at migrating this change down to the lower levels.